File: 06_Searching/01_python_syntax/02_Binary_Search.py
# Binary Search

def binary_search(arr, low,high,item):
    
    if low <= high:
        mid = (low + high) // 2
        if arr[mid] == item:
            return mid
        elif arr[mid] > item:
            return binary_search(arr, low, mid-1,item)
        else:
            return binary_search(arr,mid+1, high, item)
    else:
        return -1

arr = [10,20,30,40,50,60,70,80,90,99,100,150]

low = 0
high = len(arr)-1

print(binary_search(arr,low,high,70))

File: 06_Searching/02_C++_syntax/02_Binary_search.cpp
#include <iostream>
using namespace std;

int Binary_search(int arr[],int size,int value){
    int start = 0;
    int end = size-1;
    

    while(start<=end){
         int mid = start + (end-start)/2;
        if(arr[mid] == value){
            return mid;
        }
        else if(arr[mid]<value){
            start= mid + 1;
        }
        else{
            end = mid - 1;
        }
       

    }
    return -1;
}

int main(){
    int arr[10] = {2,3,6,7,8,9,10,15,19,20};
    int size = 10;
    int value = 20;
    int index = Binary_search(arr,size,value);
    cout<<index<<endl;

}